### PR TITLE
FieldPath static class

### PR DESCRIPTION
**Pull request by mutagen-d**
# Description

Added FieldPath static class to firestore

## How to test

```bash
npx run test path
```

### CODE REVIEW

File: mocks/path.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+class Path {
+  constructor(segments) {
+    this.segments = segments;
+  }
+  compareTo(other) {
```
Comment (sbatson5) [inline mocks/path.js:5]: There is a lot of logic in this function.  Can we get a test asserting each of these cases?

**Review by sbatson5 — CHANGES_REQUESTED** [review]
Thanks so much for the submission. This looks good. I would just like a couple more tests asserting some of the logic you have in the `path` mock

File: mocks/path.js (hunk 1 above)
Comment (mutagen-d) [reply mocks/path.js:5]: Thanks for reply. I've just added tests for  `compareTo` function

File: mocks/firestore.js [hunk 2]
```diff
@@ -332,6 +336,11 @@ FakeFirestore.CollectionReference = class extends FakeFirestore.Query {
     return requestedRecords;
   }
 
+  listDocuments() {
+    mockListDocuments();
+    return Promise.resolve([new FakeFirestore.DocumentReference('abc123', this)]);
```
[the remark this answers is not in the document]
Comment (mutagen-d) [reply mocks/firestore.js]: Fixed it

**Review by sbatson5 — APPROVED** [review]
Thanks for making those updates! 🙌

**Pull request merged**